carbon/app.py

```diff
@@ -481,64 +472,48 @@ def run(self) -> None:
+                self.config["SYMPLECTIC_FTP_PASS"],
+                self.config["SYMPLECTIC_FTP_PATH"],
+                self.config["SYMPLECTIC_FTP_HOST"],
+                int(self.config["SYMPLECTIC_FTP_PORT"]),
                 self.ssl_ctx,
             )
             PipeWriter(out=fp_w).pipe(ftp_rdr).write(feed_type)
 
 
-def sns_log(f: Callable) -> Callable:
-    """AWS SNS log decorator for wrapping a click command.
-
-    This can be used as a decorator for a click command. It will wrap
-    execution of the click command in a try/except so that any exception
-    can be logged to the SNS topic before being re-raised.
-    """
-    msg_start = "[{}] Starting carbon run for the {} feed in the {} environment."
-    msg_success = "[{}] Finished carbon run for the {} feed in the {} environment."
-    msg_fail = (
-        "[{}] The following problem was encountered during the "
-        "carbon run for the {} feed in the {} environment:\n\n"
-        "{}"
-    )
-
-    @click.pass_context
-    def wrapped(ctx: Context, *args: str, **kwargs: str) -> Callable:
-        sns_id = ctx.params.get("sns_topic")
-        if sns_id:
-            client = boto3.client("sns")
-            stage = ctx.params.get("ftp_path", "").lstrip("/").split("/")[0]
-            feed = ctx.params.get("feed_type", "")
-            client.publish(
-                TopicArn=sns_id,
-                Subject="Carbon run",
-                Message=msg_start.format(datetime.now(tz=UTC).isoformat(), feed, stage),
-            )
-            try:
-                res = ctx.invoke(f, *args, **kwargs)
-            except Exception as e:
-                client.publish(
-                    TopicArn=sns_id,
-                    Subject="Carbon run",
-                    Message=msg_fail.format(
-                        datetime.now(tz=UTC).isoformat(), feed, stage, e
-                    ),
-                )
-                raise
-            else:
-                client.publish(
-                    TopicArn=sns_id,
-                    Subject="Carbon run",
-                    Message=msg_success.format(
-                        datetime.now(tz=UTC).isoformat(), feed, stage
-                    ),
-                )
-        else:
-            res = ctx.invoke(f, *args, **kwargs)
-        return res
-
-    return update_wrapper(wrapped, f)
+def sns_log(
+    config_values: dict[str, Any], status: str, error: Exception | None = None
+) -> None:
+    sns_client = boto3.client("sns")
+    sns_id = config_values.get("SNS_TOPIC")
+    stage = config_values.get("SYMPLECTIC_FTP_PATH", "").lstrip("/").split("/")[0]
+    feed = config_values.get("FEED_TYPE", "")
+
+    if status == "start":
+        msg_start = "[{}] Starting carbon run for the {} feed in the {} environment."
+        sns_client.publish(
+            TopicArn=sns_id,
+            Subject="Carbon run",
+            Message=msg_start.format(datetime.now(tz=UTC).isoformat(), feed, stage),
```

Comment:
@ehanson8 I updated these lines of codes to use f-strings instead. Multiline f-strings will look like this: 
(Note: I had to manually format each line to follow the set line length because neither `ruff` or `black` can format f-strings properly.)

```
Message=(
                f"[{datetime.now(tz=UTC).isoformat()}] Starting carbon run for the "
                f"{feed} feed in the {stage} environment."
            )

Message=(
                f"[{datetime.now(tz=UTC).isoformat()}] Finished carbon run for the "
                f"{feed} feed in the {stage} environment."
            )

Message=(
                f"[{datetime.now(tz=UTC).isoformat()}] The following problem was "
                f"encountered during the carbon run for the {feed} feed "
                f"in the {stage} environment: {error}"
            )
```

If this looks okay to you, I can push this change!

Reply:
@ehanson8 Please see the latest commit!